includes/class-freemius.php

```diff
@@ -8272,7 +8298,7 @@ function _uninstall_plugin_event( $check_user = true ) {
          * @return string
          */
         function premium_plugin_basename() {
-            return "{$this->_slug}-premium/" . basename( $this->_free_plugin_basename );
+            return "{$this->get_premium_slug()}/" . basename( $this->_free_plugin_basename );

```

Comment:
@fajardoleo worth checking if we can change this to:
```
    return ( $this->is_premium() ) ?
        $this->_plugin_basename :
        "{$this->get_premium_slug()}/" . basename( $this->_free_plugin_basename );
```

I hope that this will protect against situations where renaming the folder of the premium version. I think that we'll need to enrich `is_premium()` for cases where `$this->_plugin->is_premium` is not set.